lgc/include/lgc/state/TargetInfo.h

```diff
@@ -77,6 +77,9 @@ struct GpuProperty {
     unsigned diffSignedness : 1; // Whether the components of two vectors have the diff signedness
   } supportIntegerDotFlag;       // The flag indicates the HW supports integer dot product
   unsigned supportsXnack;        // GPU supports XNACK
+  bool supportsDpp;              // GPU supports DPP
+  bool supportsDppRowXmask;      // GPU supports DPP ROW_XMASK
+  bool supportsPermLaneDpp;      // GPU supports perm lane DPP
```

Comment:
What exactly does this refer to?

Reply:
supportsDpp replaces https://github.com/GPUOpen-Drivers/llpc/blob/bd43984d072ba70844bbb9f35c8b99b6211370e4/lgc/builder/BuilderImpl.cpp#L243

supportsDppRowXmask replaces https://github.com/GPUOpen-Drivers/llpc/blob/bd43984d072ba70844bbb9f35c8b99b6211370e4/lgc/builder/BuilderImpl.cpp#L249

supportsPermLaneDpp replaces https://github.com/GPUOpen-Drivers/llpc/blob/bd43984d072ba70844bbb9f35c8b99b6211370e4/lgc/builder/BuilderImpl.cpp#L265

They're all used in subgroup builder - for instance here: https://github.com/GPUOpen-Drivers/llpc/blob/bd43984d072ba70844bbb9f35c8b99b6211370e4/lgc/builder/SubgroupBuilder.cpp#L966-L970

I don't know the hardware reason for why a dppmov is preferred if it's available though.

Reply:
I should have been clearer, I was trying to ask specifically about `supportPermLaneDpp`. It seems this is used to guard the use of `v_permlane[x]16_b32`, which is not DPP. DPP specifically refers to the lane permutations that are integrated into other VALU instructions via the DPP (DPP8 or DPP16) encodings.

So, this is a pre-existing problem, but still, please rename this to `supportPermLane16` for clarity. (We don't need separate 16 and x16 flags.)

Reply:
Fixed